### PR TITLE
test(e2e): cut down on resource name lengths for NRG limit

### PR DESCRIPTION
**Pull request by charliedmcb**
Fixes # 

**Description**
When testing in some new regions I noticed we were hitting some extra long NRG names:
https://github.com/Azure/karpenter/actions/runs/6950526005/job/18910870443
image

Could add something more sophisticated to ensure this is never hit, but I think trimming it down by this much is a quick, and reasonable fix.

**How was this change tested?**
Nothing additional
*

**Does this change impact docs?**
- [ ] Yes, PR includes docs updates
- [ ] Yes, issue opened: # 
- [x] No

**Release Note**

```release-note
NONE
```

### CODE REVIEW

**Review by charliedmcb — COMMENTED** [review]
/test

File: .github/workflows/e2e.yaml [hunk 1]
```diff
@@ -83,7 +83,7 @@ jobs:
           sleep $(( $RANDOM % 60 + 1 ))
       - name: generate cluster name
         run: |
-          CLUSTER_NAME=$(echo ${{ inputs.suite }}-mc-${{ inputs.hash }} | awk '{print tolower($0)}')
+          CLUSTER_NAME=$(echo mc | awk '{print tolower($0)}')
```
Comment (tallaxes) [inline .github/workflows/e2e.yaml:86]:
```suggestion
          CLUSTER_NAME=mc
```
equivalent?

**Review by charliedmcb — COMMENTED** [review]
/test

**Review by tallaxes — APPROVED** [review]

**Review by charliedmcb — COMMENTED** [review]
/test

**Comment by charliedmcb** [comment]
This is the E2E run:
https://github.com/Azure/karpenter/actions/runs/7227939042

**Comment by charliedmcb** [comment]
> This is the E2E run: https://github.com/Azure/karpenter/actions/runs/7227939042

Note: the GPU test is failing, but due to quota issue. Does not seem to be related

**Pull request merged**